src/NullConnection.php

```diff
@@ -0,0 +1,235 @@
+<?php
+/**
+ *
+ * This file is part of Atlas for PHP.
+ *
+ * @license http://opensource.org/licenses/mit-license.php MIT
+ *
+ */
+declare(strict_types=1);
+
+namespace Atlas\Query;
+
+use Atlas\Pdo\Connection;
+use BadMethodCallException;
+
+class NullConnection extends Connection
+{
+    protected $driverName;
+
+    public static function new(...$args) : Connection
+    {
+        return new static(...$args);
+    }
+
+    public function __construct(string $driverName = 'null')
+    {
+        $this->driverName = $driverName;
+    }
+
+    public function __call(
+        string $method,
+        array $params
+    ) {
+        throw $this->badMethodCallException(__METHOD__);
+    }
+
+    public function getDriverName() : string
+    {
+        return $this->driverName;
+    }
+
+    public function getPdo() : PDO
```

Comment:
This call throws an exception: `Could not check compatibility between Atlas\Query\NullConnection::getPdo(): Atlas\Query\PDO and Atlas\Pdo\Connection::getPdo(): PDO, because class Atlas\Query\PDO is not available`. It should be `\PDO`.